cpp-package/include/mxnet-cpp/.gitignore

```diff
@@ -0,0 +1,2 @@
+# Rebuildable file(s)
+op.h
```

Comment:
I am a little confused. The expectation is that someone will check out the mxnet source tree and include directly from mxnet/cpp-package/include, but dynamically load the libmxnet.so that came with their pip install rather than one that they built locally?  That seems unlikely and dangerous, correct?  
The cmake build "distributes" mxnet headers during a build if INSTALL_INCLUDE_DIR is defined as a path, although not yet for cpp-package.  If we're going to make a distribution like apt-get install mxnet-dev, I would expect that we can deploy the built op.h

wdyt?